include/phasar/DB/ProjectIRDB.h

```diff
@@ -90,16 +90,39 @@ class ProjectIRDB {
   void addFunctionToDB(const llvm::Function *F);
 
 public:
+  static llvm::PassManager<llvm::Module> createDefaultPassManager();
+  static constexpr IRDBOptions DefaultIRDBOptionsIRFiles =
+      IRDBOptions::WPA | IRDBOptions::OWNS;
+  static constexpr IRDBOptions DefaultIRDBOptionsModules = IRDBOptions::WPA;
+  static constexpr llvm::Linker::Flags DefaultLinkerFlags =
+      llvm::Linker::LinkOnlyNeeded;
+
   /// Constructs an empty ProjectIRDB
-  ProjectIRDB(IRDBOptions Options);
+  ProjectIRDB(IRDBOptions Options, llvm::PassManager<llvm::Module> Passes =
+                                       createDefaultPassManager());
+
   /// Constructs a ProjectIRDB from a bunch of LLVM IR files
   ProjectIRDB(const std::vector<std::string> &IRFiles,
-              IRDBOptions Options = (IRDBOptions::WPA | IRDBOptions::OWNS),
-              llvm::Linker::Flags LinkerFlags = llvm::Linker::LinkOnlyNeeded);
+              llvm::PassManager<llvm::Module> Passes);
+
+  ProjectIRDB(const std::vector<std::string> &IRFiles, IRDBOptions Options,
+              llvm::PassManager<llvm::Module> Passes);
+
+  ProjectIRDB(
+      const std::vector<std::string> &IRFiles,
+      IRDBOptions Options = DefaultIRDBOptionsIRFiles,
+      llvm::Linker::Flags LinkerFlags = DefaultLinkerFlags,
+      llvm::PassManager<llvm::Module> Passes = createDefaultPassManager());
+
   /// Constructs a ProjecIRDB from a bunch of LLVM Modules
-  ProjectIRDB(const std::vector<llvm::Module *> &Modules,
-              IRDBOptions Options = IRDBOptions::WPA,
-              llvm::Linker::Flags LinkerFlags = llvm::Linker::LinkOnlyNeeded);
+  ProjectIRDB(const std::vector<llvm::Module *> &Modules, IRDBOptions Options,
+              llvm::Linker::Flags LinkerFlags);
+
+  ProjectIRDB(
+      const std::vector<llvm::Module *> &Modules,
+      IRDBOptions Options = DefaultIRDBOptionsModules,
```

Comment:
Here having `IRDBOptions::OWNS` by default is not a good idea. We would then attempt to free the modules when destructing the IRDB.